components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java

```diff
@@ -591,23 +591,26 @@ private static JsonElement retrieveValidatedDTOObject(String pathToArchive, Bool
     }
 
     @NotNull
-    private static JsonObject retrievedAPIDtoJson(String pathToArchive) throws IOException {
+    private static JsonObject retrievedAPIDtoJson(String pathToArchive) throws IOException, APIManagementException {
         // Get API Definition as JSON
         String jsonContent =
                 getFileContentAsJson(pathToArchive + ImportExportConstants.API_FILE_LOCATION);
         if (jsonContent == null) {
-            throw new IOException("Cannot find API definition. api.yaml or api.json should present");
+            throw new APIManagementException("Cannot find API definition. api.yaml or api.json should present",
+                    ExceptionCodes.ERROR_FETCHING_DEFINITION_FILE);
         }
         return processRetrievedDefinition(jsonContent);
     }
 
     @NotNull
-    private static JsonObject retrievedAPIProductDtoJson(String pathToArchive) throws IOException {
+    private static JsonObject retrievedAPIProductDtoJson(String pathToArchive)
+            throws IOException, APIManagementException {
```

Comment:
Since we have replaced the IOException with APIMException do we need throw IOException now?

Reply:
The IOException is still throwing by **processRetrievedDefinition** function inside **retrievedAPIProductDtoJson**. So we need to keep it as it is.